Python/ceval.c

```diff
@@ -294,8 +308,9 @@ PyEval_RestoreThread(PyThreadState *tstate)
 
     int err = errno;
     take_gil(tstate);
+    interp->finalizing = 1;
     /* _Py_Finalizing is protected by the GIL */
-    if (_Py_IsFinalizing() && !_Py_CURRENTLY_FINALIZING(tstate)) {
+    if (interp->finalizing && !_Py_CURRENTLY_FINALIZING(tstate)) {
```

Comment:
Please check `_Py_CURRENTLY_FINALIZING()`.  I suspect that it doesn't do the right thing.  It may only matter for the main interpreter (e.g. during *runtime* finalization).

Reply:
Am abit confused, Does this mean change to instead check for _Py_CURRENTLY_FINALIZING() not !_Py_CURRENTLY_FINALIZING() ?